### ISSUE TITLE
Spec `v3` Pact files cannot be merged when publishing

### ISSUE THREAD

**Issue by martinslota** [pact-foundation/pact-js-core#439]
### Software versions

- **OS**: Mac OS 13.2.1
- **Consumer Pact library**:  PactJS v 11.0.2
- **Provider Pact library**: N/A
- **Node Version**: v18.15.0

### Issue Checklist

Please confirm the following:

- [X] I have upgraded to the latest
- [X] I have the read the FAQs in the Readme
- [X] I have triple checked, that there are **no unhandled promises** in my code
- [X] I have set my log level to debug and attached a log file showing the complete request/response cycle
- [X] For bonus points and virtual high fives, I have created a reproduceable git repository (see below) to illustrate the problem

### Expected behaviour

Several `v3` Pact files can be merged and published using the `Publisher`.

### Actual behaviour

Merging fails because the merger seems to assume to be merging spec `v2` Pact files.

### Steps to reproduce

See https://github.com/martinslota/pact-js-bug-cannot-merge-v3-pacts.

### Relevant log files

[failure.log](https://github.com/pact-foundation/pact-js-core/files/11094112/failure.log)

**Comment by mefellows** [comment]
Thanks for the repro.

I think the issue here is the ruby pact broker client that detected two contracts for the same provider and is merging themz but doesn't know how to do so with pact specs > version 2.

The solution here is not to write to two separate pact files and merge on publish. It's safe to point both providers at the same output directory 

Could you please see if that works?

**Comment by martinslota** [comment]
> I think the issue here is the ruby pact broker client that detected two contracts for the same provider and is merging themz but doesn't know how to do so with pact specs > version 2.

Yes, I believe it is [this script](https://github.com/pact-foundation/pact_broker-client/blob/master/lib/pact_broker/client/merge_pacts.rb) that only really works for spec version 2.

> The solution here is not to write to two separate pact files and merge on publish. It's safe to point both providers at the same output directory
> 
> Could you please see if that works?

We parallelize our consumer tests on several workers that each produce a Pact file. We do need to merge those files somehow to publish a single pact for that version of the consumer. Also, duplicate removal is really cool to have in our case. If we merge the fails in a naive way (just concatenating the `interactions` arrays), then the pact file has lots of duplicates and verification of the provider becomes a lot slower.

**Comment by martinslota** [comment]
I would also argue that since the `Publisher` takes an argument called `pactFilesOrDirs`, it should be able to deal with more than one file.

**Comment by mefellows** [comment]
> 
I would also argue that since the Publisher takes an argument called pactFilesOrDirs, it should be able to deal with more than one file.

The usual practice here would be to have 1 file per _provider_. So a consumer can have multiple providers, each contained within a single pact file.

The bug here is upstream as you said. But it is easily worked around by not producing more than one pact file per provider.
